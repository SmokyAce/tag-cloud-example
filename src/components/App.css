.App {
    font-family: Lucida Console, Courier, monospace;
    text-align: center;
    display: flex;
    flex-direction: column;
    height: 100vh;
}

.text-center {
    text-align: center
}
