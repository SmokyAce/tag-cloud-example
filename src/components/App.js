import React, { lazy, Suspense } from 'react';
import {
  BrowserRouter as Router,
  Route,
  Redirect,
  Switch
} from 'react-router-dom';
import Spinner from './Spinner';
import './App.css';

const HomePage = lazy(() => import('./HomePage'));
const TagPage = lazy(() => import('./TagPage'));
const NotFoundPage = lazy(() => import('./NotFoundPage'));

const App = () => {
  return (
    <div className='App'>
      <Router>
        <Suspense fallback={<Spinner />}>
          <Switch>
            <Route exact path='/(home)?' component={() => <HomePage />} />
            <Route
              path='/home/:tagId+'
              component={props => <TagPage {...props} />}
            />
            <Route component={() => <NotFoundPage />} />
            <Redirect to='/' />
          </Switch>
        </Suspense>
      </Router>
    </div>
  );
};

export default App;
