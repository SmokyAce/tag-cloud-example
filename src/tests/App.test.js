/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import App from '../components/App';

describe('<App />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const snapshot = TestRenderer.create(<App />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
