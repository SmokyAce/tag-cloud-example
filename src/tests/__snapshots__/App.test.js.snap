// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="header"
  >
    <h1>
      Tag Cloud
    </h1>
  </div>
  <div
    className="cloud-container"
  >
    <div
      id="loading"
    />
    <div
      id="word-cloud"
    />
  </div>
</div>
`;
